### PR TITLE
feat: add support to create hmac key

### PR DESCRIPTION
**Pull request by jor2**
### Description

Replace this text with a summary of the changes in this PR. Include why the changes are needed and context about the changes. List required dependencies.

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] The PR references a GitHub issue.
- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -23,6 +23,24 @@ variable "create_cos_instance" {
...
 }
 
+variable "create_hmac_key" {
+  description = "Set as true to create a new HMAC key for the Cloud Object Storage instance."
+  type        = bool
+  default     = true
+}
+
+variable "hmac_key_name" {
+  description = "Set as true to create a new HMAC key for the Cloud Object Storage instance."
```
Comment (ocofaigh) [inline variables.tf:33]: incorect description

File: variables.tf [hunk 2]
```diff
@@ -23,6 +23,24 @@ variable "create_cos_instance" {
...
+  type        = string
+  default     = "my-key"
```
Comment (ocofaigh) [inline variables.tf:35]: maybe default to `hmac-cos-key`

File: variables.tf [hunk 3]
```diff
@@ -23,6 +23,24 @@ variable "create_cos_instance" {
...
+}
+
+variable "hmac_key_role" {
+  description = "Set as true to create a new HMAC key for the Cloud Object Storage instance."
```
Comment (ocofaigh) [inline variables.tf:39]: incorect description

**Review by ocofaigh — CHANGES_REQUESTED** [review]
See comments

File: variables.tf (hunk 1 above)
Comment (ocofaigh) [reply variables.tf:33]: marking as unresolved, not addressed yet

Comment (jor2) [reply variables.tf]: updated, was just marking off as i did them locally to not get confused. will just leave unresolved next time.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**